### DISCUSSION TITLE
How to expose base methods?

### DISCUSSION THREAD

**Discussion by domcir** [bytedeco/javacpp#618] [General]
Hi

I had to disable exposing a base class but I would like to expose some methods from that class, how can I do that?
Example
```java
        class A{
            public:
                int a{0};
                A(int _a) : a(_a) { }
               int getA() { return a; }
        };

        class B : public A{
            public:
                int b
                B(int _a, int _b): A(_a), b{_b}
                { }
               int getB() { return b; }
        };
```
and when using:
```java
.put(new Info("B").base("Pointer").pointerTypes("B").define())
.put(new Info("A).skip())
```

The getA() method isn't exposed. I know I could remove the base definition but in my library I can't do that as it's one of the boost classes with a lot of methods that can't be exposed.

Thanks
Dominik

**Comment by saudet** [comment]
We can get it to put custom Java code for that kind of use case:
https://github.com/bytedeco/javacpp/wiki/Mapping-Recipes#mapping-a-declaration-to-custom-code

**Reply by domcir** [reply]
Hi
Could you define the syntax for my example, please?

I've been tring different compbinations but no luck so far...
I tried something like:
`infoMap.put(new Info("B::getA()").javaText("public native @ByVal int getA();"));`
Thanks
Dominik

**Reply by saudet** [reply]
You'll need to pick something that is part of the class B. If it has a destructor, for example, by default nothing gets output, but we can use it to tell JavaCPP where to output the javaText.

**Reply by domcir** [reply]
Thanks for a quick response. It works but obviously I need to have something in my chilld clllass I can use. Also woullld a generate cpp jni fille automatically link approporiate functions?

**Reply by saudet** [reply]
If there's no destructor, you can use any other members and output all that you need there. There is no limitations as to what we can modify with javaText(). As to whether what you put there generates JNI that compiles correctly, that's going to depend on what you put there, obviously, so give it a try and see!

**Reply by domcir** [reply]
Your solution works, thanks!
